ast/annotations.go

```diff
@@ -686,13 +705,7 @@ func (as *AnnotationSet) Flatten() []*AnnotationsRef {
 
 	// Sort by path, then annotation location, for stable output
 	sort.SliceStable(refs, func(i, j int) bool {
-		if refs[i].Path.Compare(refs[j].Path) < 0 {
-			return true
-		}
-		if refs[i].Annotations.Location.Compare(refs[j].Annotations.Location) < 0 {
-			return true
-		}
-		return false
+		return refs[i].Compare(refs[j]) < 0
```

Comment:
In the  new `AnnotationsRef.Compare` method we have an extra check `ar.Annotations.Compare(other.Annotations)` in addition to the path and location comparison. Will that have an impact on the way this current check behaves where we simply return `false`?

Reply:
In any real-world scenario, the comparison would never fall through to comparing the actual annotations here, as annotations would always have a differing `Location`. On the `Insert` side, though, we might be inserting annotations coming from the bundle's `.manifest`, where loc info is stripped; at which point we want to make sure that we always get a stable order in cases where the `entrypoint` is equal (e.g. annotations with different scope, but for the same rule or package). I wanted the `Flatten`- and `Insert` sorting to use the exact same algorithm for stability clarity.
Should my assumption be off, or be changed by some future update, worst-case, we now have extra stability in sorting output; which is actually more of a feature than a bug 😄 .